okapi-core/src/main/java/org/folio/okapi/managers/TimerManager.java

```diff
@@ -78,6 +78,89 @@ private void tenantChange(String tenantId) {
     startTimers(tenantId, false);
   }
 
+  private Future<Void> loadFromStorage(String tenantId) {
+    final LockedTypedMap1<TimerDescriptor> timerMap = tenantTimers.get(tenantId);
+    return timerStore.getAll().compose(list -> {
+          List<Future<Void>> futures = new LinkedList<>();
+          int prefixLen = tenantId.length() + 1;
+          for (TimerDescriptor timerDescriptor : list) {
+            String tenantTimerId = timerDescriptor.getId();
+            if (tenantTimerId.startsWith(tenantId + "_")) {
+              timerDescriptor.setId(tenantTimerId.substring(prefixLen));
+              if (timerDescriptor.isModified()) {
+                futures.add(timerMap.put(timerDescriptor.getId(), timerDescriptor));
+              }
+            }
+          }
+          return GenericCompositeFuture.all(futures).mapEmpty();
+        }
+    );
+  }
+
+  private Future<Void> removeStale(String tenantId, List<ModuleDescriptor> mdList) {
+    final LockedTypedMap1<TimerDescriptor> timerMap = tenantTimers.get(tenantId);
+    return timerMap.getAll().compose(list -> {
+      List<Future<Void>> futures = new LinkedList<>();
+      for (String timerId : list.keySet()) {
+        futures.add(getModuleForTimer(mdList, timerId).compose(md -> {
+          if (md != null) {
+            return Future.succeededFuture();
+          }
+          final String runId = tenantId + TIMER_ENTRY_SEP + timerId;
+          if (timerRunning.containsKey(runId)) {
+            vertx.cancelTimer(timerRunning.get(runId));
+            timerRunning.remove(runId);
+          }
```

Comment:
```suggestion
          Long id = timerRunning.remove(runId);
          if (id != null) {
            vertx.cancelTimer(id);
           }
```
Only a single call to timerRunning is needed.